### PR TITLE
tracer: fix data race, start tracer before task restore

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
fix data race
```
WARNING: DATA RACE
Read at 0x00c0000e30c8 by goroutine 130:
  github.com/pingcap/dm/pkg/tracing.(*Tracer).AddJob()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/pkg/tracing/tracer.go:232 +0x1c4
  github.com/pingcap/dm/pkg/tracing.(*Tracer).CollectSyncerBinlogEvent()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/pkg/tracing/tracer_syncer.go:54 +0x3c5
  github.com/pingcap/dm/syncer.(*Syncer).Run()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/syncer/syncer.go:1156 +0x20d3
  github.com/pingcap/dm/syncer.(*Syncer).Process()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/syncer/syncer.go:473 +0x7b1

Previous write at 0x00c0000e30c8 by goroutine 29:
  github.com/pingcap/dm/pkg/tracing.(*Tracer).newJobChans()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/pkg/tracing/tracer.go:158 +0x7e
  github.com/pingcap/dm/pkg/tracing.(*Tracer).Start()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/pkg/tracing/tracer.go:81 +0x280
  github.com/pingcap/dm/dm/worker.(*Worker).Start()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/worker.go:125 +0x694
  github.com/pingcap/dm/dm/worker.(*Server).Start.func1()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/server.go:83 +0x96

Goroutine 130 (running) created at:
  github.com/pingcap/dm/dm/worker.(*SubTask).Run()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/subtask.go:159 +0x44f
  github.com/pingcap/dm/dm/worker.(*Worker).StartSubTask()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/worker.go:202 +0x36d
  github.com/pingcap/dm/dm/worker.(*Worker).Start()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/worker.go:118 +0x2c6
  github.com/pingcap/dm/dm/worker.(*Server).Start.func1()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/server.go:83 +0x96

Goroutine 29 (running) created at:
  github.com/pingcap/dm/dm/worker.(*Server).Start()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/dm/worker/server.go:80 +0x285
  github.com/pingcap/dm/cmd/dm-worker.main()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/cmd/dm-worker/main.go:66 +0x3e1
  github.com/pingcap/dm/cmd/dm-worker.TestRunMain()
      /home/apple/work/code/pingcap/gopath/src/github.com/pingcap/dm/cmd/dm-worker/main_test.go:36 +0x160
  testing.tRunner()
      /home/apple/.gvm/gos/go1.12/src/testing/testing.go:865 +0x163
```
tracer initialization is in DM-worker `Start` procedure, if task started before tracer job channels' initialization, data race happens

### What is changed and how it works?

* start tracer first, then restore task.
* refused to add tracing job if job channels are not ready
* some test code refine

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: tests/README.md [hunk 1]
```diff
@@ -37,6 +37,8 @@
 
     > If want to run one integration test case only, just pass the CASE parameter, such as `make integration_test CASE=sharding`.
 
+    > There exists some environment variables that you can configuratio by yourself, including `MYSQL_HOST1`, `MYSQL_PORT1`, `MYSQL_HOST2`, `MYSQL_PORT2`, `RESET_MASTER`. If `RESET_MASTER` is not set or set to true, `RESET MASTER` will be executed at upstream MySQL before each case.
```
Comment (csuzhangxc) [inline tests/README.md]:
```suggestion
    > There exists some environment variables that you can set by yourself, including `MYSQL_HOST1`, `MYSQL_PORT1`, `MYSQL_HOST2`, `MYSQL_PORT2`, `RESET_MASTER`. If `RESET_MASTER` is not set or set to true, `RESET MASTER` will be executed at upstream MySQL before each case.
```
?

File: pkg/tracing/tracer.go [hunk 2]
```diff
@@ -224,6 +224,10 @@ func (t *Tracer) jobProcessor(ctx context.Context, jobChan <-chan *Job) {
 
 // AddJob add a job to tracer
 func (t *Tracer) AddJob(job *Job) {
+	if t.jobsClosed.Get() {
```
Comment (GregoryIan) [inline pkg/tracing/tracer.go]: does it really resolve data race problem?

Comment (amyangfei) [reply pkg/tracing/tracer.go]: yep, if we want to avoid race condition without using `Lock`, we must ensure initial Tracer before any job operation

Comment (amyangfei) [reply pkg/tracing/tracer.go]: So we only set `jobsClosed` to false at the end of `newJobChans`, which can guarantee no access to jobs channels before they are created.

Comment (GregoryIan) [reply pkg/tracing/tracer.go]: there are still temporal logic conflict problem

Comment (amyangfei) [reply pkg/tracing/tracer.go]: yep, it is not good design, I add lock to protect the access to jobs channel

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**